Paintroid/src/androidTest/java/org/catrobat/paintroid/test/espresso/MenuFileActivityIntegrationTest.kt

```diff
@@ -0,0 +1,554 @@
+        onView(withId(R.id.pocketpaint_save_dialog_spinner)).perform(click())
+        onData(allOf(`is`(instanceOf<Any>(String::class.java)), `is`<String>("png")))
+            .inRoot(RootMatchers.isPlatformPopup()).perform(click())
+        runBlocking {
+            onView(withText(R.string.save_button_text)).perform(click())
+            delay(100)
+        }
+        onTopBarView().performOpenMoreOptions()
+        onView(withText(R.string.menu_save_copy)).perform(click())
+        imageNumber = launchActivityRule.activity.presenter.imageNumber
+        onView(withText("png")).check(matches(isDisplayed()))
+        onView(withText("image$imageNumber")).check(matches(isDisplayed()))
+    }
+
+    private fun createImageIntent() {
+        val intent = Intent()
+        intent.data = createTestImageFile()
+        val resultOK = ActivityResult(Activity.RESULT_OK, intent)
+        Intents.intending(hasAction(Intent.ACTION_GET_CONTENT)).respondWith(resultOK)
+    }
+
+    private fun createTestImageFile(): Uri? {
+        val bitmap = Bitmap.createBitmap(400, 400, Bitmap.Config.ARGB_8888)
+        val contentValues = ContentValues()
+        contentValues.put(MediaStore.Images.Media.DISPLAY_NAME, "testfile.jpg")
+        contentValues.put(MediaStore.Images.Media.MIME_TYPE, "image/jpeg")
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.Q) {
+            contentValues.put(MediaStore.Images.Media.RELATIVE_PATH, Environment.DIRECTORY_PICTURES)
+        }
+        val resolver = activity.contentResolver
+        val imageUri = resolver.insert(MediaStore.Images.Media.EXTERNAL_CONTENT_URI, contentValues)
+        try {
+            val fos = imageUri?.let { resolver.openOutputStream(it) }
+            assertTrue(bitmap.compress(Bitmap.CompressFormat.JPEG, 100, fos))
+            assert(fos != null)
+            fos?.close()
+        } catch (e: IOException) {
+            throw AssertionError("Picture file could not be created.", e)
+        }
+        val imageFile = File(imageUri?.path, "testfile.jpg")
+        deletionFileList.add(imageFile)
+        return imageUri
+    }
+
+    @Test
+    fun testLoadImageTransparency() {
+        val intent = Intent()
+        intent.data = createTestImageFile()
+        val result = ActivityResult(Activity.RESULT_OK, intent)
+        Intents.intending(hasAction(Intent.ACTION_GET_CONTENT)).respondWith(result)
+        onTopBarView().performOpenMoreOptions()
+        onView(withText(R.string.menu_load_image)).perform(click())
+        ToolBarViewInteraction.onToolBarView().performSelectTool(ToolType.ERASER)
+        onDrawingSurfaceView().checkPixelColor(Color.BLACK, BitmapLocationProvider.MIDDLE)
+        onDrawingSurfaceView().perform(touchAt(MIDDLE))
+        onDrawingSurfaceView().checkPixelColor(Color.TRANSPARENT, BitmapLocationProvider.MIDDLE)
+    }
+
+    private fun addUriToDeletionFileList(uri: Uri?) {
+        uri?.path?.let {
+            deletionFileList.add(File(it))
+        }
+    }
+}
```

Comment:
Next time, please make an extra PR for the tests ;)

Reply:
sure :)